### PR TITLE
[Feature] Polish tutorials & misc updates

### PR DESCRIPTION
**Pull request by wentww**
- Accomplish tutorials
	- 01_get_started.ipynb
	- 02_randomization.ipynb
	- 03_human_nerf.ipynb

- Actor
	- add properties: dimensions and bound_box
	- complete TODOs

- Add `xrfeitoria/utils/reader.py` for loading images and annotations.
	- add `pip install xrfeitoria[vis]`

### CODE REVIEW

File: xrfeitoria/factory.py [hunk 1]
```diff
@@ -200,6 +200,8 @@ def __init__(
         )
         self._rpc_runner.start()
         self.utils.init_scene_and_collection(default_level_blender, self._cleanup)
+        self.utils.set_env_color(color=(1.0, 1.0, 1.0, 1.0))
+        self.engine = 'blender'
```
Comment (HaiyiMei) [inline xrfeitoria/factory.py:204]: delete this property. use `_tls.cache['platform'].name`

**Review by HaiyiMei — CHANGES_REQUESTED** [review]

File: tutorials/02_add_transform_keys.ipynb [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    "The following code renders all the render jobs and save the images to the ``output_path`` set in ``seq.add_to_renderer`` above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xf_runner.render()"
```
Comment (HaiyiMei) [inline tutorials/02_add_transform_keys.ipynb:287]: `## 5. Render` and `## 6. Close the engine` are duplicates in tutorial01. Please consider squashing them into one cell.

Comment (HaiyiMei) [reply tutorials/02_add_transform_keys.ipynb:287]: Plus, consider renaming this tutorial to something like 02_randomization.

File: tutorials/01_get_started.ipynb [hunk 3]
```diff
@@ -8,14 +8,18 @@
     "\n",
     "## Overview\n",
     "\n",
-    "🚀 In this tutorial, we'll initialize XRFeitoria and add basic shapes."
+    "🚀 This tutorial demonstrates a minimal application of XRFeitoria. By the end of this tutorial, you will be able to:\n",
+    "- Open Blender/Unreal Engine by XRFeitoria\n",
+    "- Import a mesh\n",
+    "- Add a camera\n",
+    "- Render the mesh and save the image"
```
Comment (HaiyiMei) [inline tutorials/01_get_started.ipynb:15]: I prefer to stress the essential steps that are needed to complete this MVP.

```suggestion
    "🚀 This tutorial demonstrates a minimal application of XRFeitoria. To accomplish this, essential steps should be taken:
    - initialization
    - import an `actor`
    - add a `camera`
    - add a `sequence`
    - render
    
    [`actor`](https://xrfeitoria.readthedocs.io/en/latest/apis/actor.html)
, [`camera`](https://xrfeitoria.readthedocs.io/en/latest/apis/camera.html), and [`sequence`](https://xrfeitoria.readthedocs.io/en/latest/apis/sequence.html) are classes defined in `XRFeitoria`.  We'll introduce them in detail step by step.
```

File: xrfeitoria/actor/actor_unreal.py [hunk 4]
```diff
@@ -22,6 +22,11 @@ class ActorUnreal(ActorBase):
 
     _object_utils = ObjectUtilsUnreal
 
+    @property
+    def engine_path(self) -> Vector:
```
Comment (HaiyiMei) [inline xrfeitoria/actor/actor_unreal.py:26]:
```suggestion
    def engine_path(self) -> str:
```

File: xrfeitoria/actor/actor_blender.py [hunk 5]
```diff
@@ -31,7 +31,8 @@ class ActorBlender(ActorBase):
     @property
     def dimensions(self) -> Vector:
         """Dimensions of the actor."""
-        return self._object_utils.get_dimensions(self.name)
+        bbox_min, bbox_max = self.bound_box
+        return (bbox_max[0] - bbox_min[0], bbox_max[1] - bbox_min[1], bbox_max[2] - bbox_min[2])
```
Comment (HaiyiMei) [inline xrfeitoria/actor/actor_blender.py:35]: move this to `ObjectUtilsBlender`

File: xrfeitoria/object/object_utils.py [hunk 6]
```diff
@@ -247,19 +271,6 @@ class ObjectUtilsBlender(ObjectUtilsBase):
...
-            name (str): Name of the object.
-
-        Returns:
-            Vector: Dimensions of the object.
-        """
-        cls.validate_name(name)
-        return cls._get_dimensions_in_engine(name=name)
-
     @classmethod
     def get_bbox(cls, name: str) -> 'Dict[Tuple[Vector, Vector]]':
```
Comment (HaiyiMei) [inline xrfeitoria/object/object_utils.py:275]: `get_bbox` and `get_bound_box` seem to be duplicated, as well as `_get_bbox_in_engine` and `_get_bound_box_in_engine`.

File: xrfeitoria/renderer/renderer_unreal.py [hunk 7]
```diff
@@ -155,6 +155,9 @@ def render_jobs(cls) -> None:
         # post process, including: convert cam params.
         cls._post_process()
 
+        # clear render queue
+        cls.clear()
+
```
Comment (HaiyiMei) [inline xrfeitoria/renderer/renderer_unreal.py:160]: - add this clear info in the docstring of this function (also applicable for renderer_blender)
- delete line 152

File: xrfeitoria/sequence/sequence_blender.pyi [hunk 8]
```diff
@@ -11,16 +11,6 @@ class SequenceBlender(SequenceBase):
...
-    def import_actor(
-        cls,
-        file_path: PathLike,
-        actor_name: Optional[str] = ...,
-        location: Vector = ...,
-        rotation: Vector = ...,
-        scale: Vector = ...,
-        stencil_value: int = ...,
-    ) -> ActorBlender: ...
-    @classmethod
```
Comment (HaiyiMei) [inline xrfeitoria/sequence/sequence_blender.pyi:23]: - This should not be deleted. The method of inherited class should have a unique typing hint.
- A similar hint in sequence_unreal is needed as well

File: tutorials/02_add_transform_keys.ipynb [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+    "    # In render job, you can specify the output path, resolution, render passes, etc.\n",
+    "    # The ``output_path`` is the path to save the rendered data.\n",
+    "    # The ``resolution`` is the resolution of the rendered image.\n",
+    "    # The ``render_passes`` define what kind of data you want to render, such as img, depth, normal, etc.\n",
+    "    # and what kind of format you want to save, such as png, exr, etc.\n",
+    "    seq.add_to_renderer(\n",
+    "        output_path=f'./tutorial02/outputs/{xf_runner.engine}/',\n",
+    "        resolution=(1280, 720),\n",
+    "        render_passes=[RenderPass('img', 'png')],\n",
+    "    )"
```
[the remark this answers is not in the document]
Comment (wentww) [reply tutorials/02_add_transform_keys.ipynb]: fixed in ce589b6

File: tutorials/02_add_transform_keys.ipynb (hunk 2 above)
Comment (wentww) [reply tutorials/02_add_transform_keys.ipynb]: 1. fixed in ce589b6
2. fixed in 4f478f9

File: tutorials/01_get_started.ipynb (hunk 3 above)
Comment (wentww) [reply tutorials/01_get_started.ipynb]: fixed in ce589b6

File: xrfeitoria/actor/actor_unreal.py (hunk 4 above)
Comment (wentww) [reply xrfeitoria/actor/actor_unreal.py]: fixed in 6a97fac

File: xrfeitoria/actor/actor_blender.py (hunk 5 above)
Comment (wentww) [reply xrfeitoria/actor/actor_blender.py]: fixed in ee5504d

File: xrfeitoria/object/object_utils.py (hunk 6 above)
Comment (wentww) [reply xrfeitoria/object/object_utils.py]: fixed in ee5504d

File: xrfeitoria/renderer/renderer_unreal.py (hunk 7 above)
Comment (wentww) [reply xrfeitoria/renderer/renderer_unreal.py:168]: fixed in e536fc2

File: xrfeitoria/sequence/sequence_blender.pyi (hunk 8 above)
Comment (wentww) [reply xrfeitoria/sequence/sequence_blender.pyi]: fixed in fba8cf2

**Review by HaiyiMei — APPROVED** [review]

**Pull request merged**